### PR TITLE
Render scans for rabbit

### PR DESCRIPTION
**Pull request by cap10morgan**
While writing integration tests for ballot-scout status events, I realized that the data structures from usps-processor were of a sufficiently different shape that ballot-scout's fns weren't processing them correctly.

I fixed this by calling the same render-scan fn used by the HTTP API on published scans. We did this in ballot-scout recently too for status events.

While I was in there anyway, I also fixed the problem of publishing events regardless of whether the scans saved in Datomic by having the `db/store-scans` fn return a seq of the Datomic scan entities it just transacted. I then only render and publish those to the event exchange.

Assigning @bwreid.

### CODE REVIEW

File: src/usps_processor/db.clj [hunk 1]
```diff
@@ -1,7 +1,8 @@
 (ns usps-processor.db
-  (:require [turbovote.datomic-toolbox :as d]))
+  (:require [turbovote.datomic-toolbox :as d]
```
Comment (bwreid) [inline src/usps_processor/db.clj]: Is it possible to revert the trend of assigning these to a single letter? Just me?

File: src/usps_processor/api.clj [hunk 2]
```diff
@@ -38,11 +38,15 @@
...
-      (select-keys [:scan/time :scan/barcode :scan/facility-zip
-                    :scan/operation-code :scan/service])
-      attach-facility-city-state))
+(defn render-scan
+  [scan & [attach-mailing]]
+  (let [scan-render (-> scan
+                        (select-keys [:scan/time :scan/barcode :scan/facility-zip
+                                      :scan/operation-code :scan/service])
+                        attach-facility-city-state)]
+    (if attach-mailing
```
Comment (bwreid) [inline src/usps_processor/api.clj]: Can we test this functionality explicitly?

**Comment by bwreid** [comment]
It'd be nice to make some sweet docs for this.

File: src/usps_processor/db.clj (hunk 1 above)
Comment (cap10morgan) [inline src/usps_processor/db.clj]: I agree. How do you feel about this convention?

`datomic` for `datomic.api`
`datomic-tb` for `turbovote.datomic-toolbox`

File: src/usps_processor/api.clj (hunk 2 above)
Comment (cap10morgan) [inline src/usps_processor/api.clj]: Yes! I'll add a test or two for that.

Comment (cap10morgan) [inline src/usps_processor/api.clj]: Actually Troy and I ended up refactoring this such that render-scan is unchanged from kraken (and master too, IIRC) and we merge in the mailing in `queue/publish-scan` instead, which feels cleaner. More commits coming soon.

**Comment by cap10morgan** [comment]
@bwreid Agreed on the docs. Do you mean specifically the event message format, or did you have other things in mind too?

**Comment by cap10morgan** [comment]
@bwreid Other than doc updates, this is ready for review again.

File: src/usps_processor/db.clj (hunk 1 above)
Comment (bwreid) [inline src/usps_processor/db.clj]: I like! Would like to know what @tie-rack thinks.

**Comment by bwreid** [comment]
@cap10morgan I think the event message format would be helpful (similar to some of the earlier docs) and it'd be nice to get a kind of example of getting it running and working. The docs already in the readme do a good job of setting it up. Seem reasonable?

Other than that and the merge conflicts. Looks nice.

**Comment by cap10morgan** [comment]
@bwreid Running it has changed quite a bit on the kraken branch w/ the introduction of docker-compose and removal of Immutant 1 (soon to be replaced w/ Immutant 2).

I'll document the ballot create and update message formats, and then when the latest kraken is merged in here (to make the PR mergeable again), do you want to take another look at the docs and see if anything seems missing?

**Comment by bwreid** [comment]
LGTM overall. However, can we please add to Torchpad more specific instructions for us? (eg. which queues to use / not use)

Merge whenever that is done and/or you have some process in place to make sure we don't forget that.

**Comment by cap10morgan** [comment]
[Submitted for your approval](https://democracyworks.torchpad.com/kraken/usps-processor), @bwreid.

**Comment by bwreid** [comment]
@cap10morgan I mean, I really feel like we need a cat/dog picture at the end there, but I suppose it'll do. :princess: :sparkles: Yay!

**Pull request merged**